replit/upm
# feat/flit support

Why
===

As we continue to expand into the realm of offering better Python package manager support, we're happy to add `flit` to upm.

What changed
============

- Bump `BurntSushi/toml` to 1.4.0
- Broke out `python/poetry.go` and `python/pip.go` to go along with `python/flit.go`
- Fix a bug with pip `requirements.txt` additions where we would take a zero-byte `requirements.txt` and turn it into `\ndep` instead of `dep\n`
- Adding `QuirksCannotAddRemove` to support `flit`, which does not directly support `add` or `remove`. In the future, we can or delete from `pyproject.toml`

Test plan
=========

- Import `https://github.com/mwaskom/seaborn` into Replit
- Run `pip install flit` to get a compatible version of flit
- Run `upm install`
- Verify that dependencies are installed
